react.js

```diff
@@ -11,13 +11,22 @@ module.exports = {
     'airbnb/rules/react',
     'airbnb/rules/react-a11y',
   ],
-
+  plugins: [
+    'react-hooks',
+  ],
   rules: {
+    /* Base React rules */
     // Allow class methods that do not use `this`.
     'class-methods-use-this': OFF,
     // Warn when a component should be specified as a stateless function if it doesn't require state/lifecycle methods.
     'react/prefer-stateless-function': OFF,
     // Warn when there's more than one expression per line in JSX.
     'react/jsx-one-expression-per-line': OFF,
+    // Allow .js files to have JSX
+    'react/jsx-filename-extension': [WARNING, { 'extensions': ['.js'] }],
+
+    /* React Hooks rules */
+    'react-hooks/rules-of-hooks': ERROR,
+    'react-hooks/exhaustive-deps': WARNING,
```

Comment:
there are few custom rules or exceptions i'd like to propose. they can be requested in another PR. looks good!

Reply:
yes please do! thanks!